template.tpl

```diff
@@ -331,51 +378,126 @@ const initWidget = () => {
+  log({ storedEvent: storedEvent });
+
+  if (!(storedEvent && storedEvent.queryId && storedEvent.placement_id)) {
+    log('Unable to retrieve last clicked widget for event: ', eventName);
+    data.gtmOnSuccess();
+    return;
+  }
+
+  for (const ev of eventsArr) {
+    ev.queryId = storedEvent.queryId;
+  }
+
+  // initialize visenzeLayer if not available
+  const vsLayerRef = copyFromWindow(VS_LAYER_REF);
+  if (!vsLayerRef) {
+    setInWindow(VS_LAYER_REF, []);
+  }
+  // call visenzeLayer.push method, with event
+  const vsLayerBody = {
+    action: 'sendEvents',
+    placementId: storedEvent.placement_id,
+    params: [eventName, eventsArr],
+  };
+  const sendEventForDeployType = shouldDeployTypeSendEvent();
+  if (!sendEventForDeployType) {
+    vsLayerBody.deployTypeId = '1';
+  }
+
+  callInWindow(vsLayerPushMethod, vsLayerBody);
+
+  log({ eventName: eventName, body: vsLayerBody, method: vsLayerPushMethod, visenzeLayer_original: vsLayerRef });
+};
 
 const sendAtcEvent = () => {
-  if (shouldSendEvent()) {
-    const EVENT_NAME = 'add_to_cart';
-    sendWidgetEvent(EVENT_NAME, { pid: productId });
-  } else {
+  const productId = getProductId();
+  if (!productId) {
+    logWithError({ msg: 'event not triggered', integrationType: data.integrationType, productId: productId });
     data.gtmOnFailure();
+    return;
   }
+
+  const EVENT_NAME = 'add_to_cart';
+  const successEv = sendWidgetEvent(EVENT_NAME, [{ pid: productId }]);
+  data.gtmOnSuccess();
 };
 
+const sendTransactionEventForProductArr = (productArr) => {
+  const EVENT_NAME = 'transaction';
+  const eventsArr = [];
+
+  for (const productEl of productArr) {
+    const pid = productEl[data.productArrSkuField];
+    if (!pid) {
+      logWithError({ msg: 'unable to find productId in product', field: data.productArrSkuField, product: productEl });
+      continue;
+    }
+
+    const qtyStr = productEl[data.productArrQtyField];
+    const priceStr = productEl[data.productArrPriceField];
+
+    if (!qtyStr || !priceStr) {
+      logWithError({ msg: 'unable to find quantity or price field', quantity_field: data.productArrQtyField, price_field: data.productArrPriceField });
+    }
+
+    const qty = makeNumber(qtyStr || '1');
+    const price = makeNumber(priceStr || '0');
+    eventsArr.push({ pid: pid, order_value: qty * price });
+  }
+
+  sendWidgetEvent(EVENT_NAME, eventsArr);
+  data.gtmOnSuccess();
+};
 
 const sendTransactionEvent = () => {
-  if (shouldSendEvent()) {
-    const EVENT_NAME = 'transaction';
-    const dataLayerArr = copyFromWindow('dataLayer');
-    const orderValueKeys = data.orderValueFieldName.split('.');
-    const orderValue = getFromDataLayers(dataLayerArr, getProductIdByNestedKey, orderValueKeys);
-
-    sendWidgetEvent(EVENT_NAME, { pid: productId, order_value: orderValue });
-  } else {
+  const dataLayerArr = copyFromWindow('dataLayer');
+  const productArrFieldKeys = data.productArrFieldName.split('.');
```

Comment:
is there a need to check if productArrFieldName is valid / define?

Reply:
added empty string coercion for empty `productArrFieldName` field